app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -71,7 +188,32 @@ class ProfileForm extends Component {
       communityConfig,
     } = this.props;
     const { profile_image_90, username, name } = this.user;
-    const { canSkip } = this.state;
+    const { canSkip, groups } = this.state;
+    let sections = [];
+
+    if (groups) {
+      sections = groups.map((group) => {
+        return (
+          <div class="onboarding-profile-sub-section">
+            <h2>{group.name}</h2>
+            {group.description && (
+              <div class="color-base-60">{group.description})</div>
+            )}
+            <div>
+              {group.profile_fields.map((field) => {
+                return field.input_type === 'check_box'
+                  ? this.checkboxField(field)
+                  : field.input_type === 'color_field'
+                  ? this.colorField(field)
+                  : field.input_type === 'text_area'
+                  ? this.textAreaField(field)
+                  : this.textField(field);
+              })}
+            </div>
+          </div>
+        );
+      });
+    }
```

Comment:
refactor (non-blocking): The conditional check on groups is not necessary if you destructure `groups` from `this.state` with a default value of `[]`.

```suggestion
    const { canSkip, groups = [] } = this.state;
    const sections = groups.map((group) => {
      return (
        <div class="onboarding-profile-sub-section">
          <h2>{group.name}</h2>
          {group.description && (
              <div class="color-base-60">{group.description})</div>
          )}
          <div>
            {group.profile_fields.map((field) => {
              return field.input_type === 'check_box'
                ? this.checkboxField(field)
                : field.input_type === 'color_field'
                ? this.colorField(field)
                : field.input_type === 'text_area'
                ? this.textAreaField(field)
                : this.textField(field);
            })}
          </div>
        </div>
        );
      });
```

Reply:
✅